av/stream.pxd

```diff
@@ -3,24 +3,22 @@ from libc.stdint cimport int64_t
 cimport libav as lib
 
 from av.codec.context cimport CodecContext
-from av.container.core cimport Container, ContainerProxy
+from av.container.core cimport Container
 from av.frame cimport Frame
 from av.packet cimport Packet
 
 
 cdef class Stream(object):
 
     # Stream attributes.
-    cdef ContainerProxy _container
-    cdef _weak_container
+    cdef readonly Container container
```

Comment:
I'm thinking we need some kind of mechanism to break the Container <-> Stream reference loop when the container gets closed, but this can probably be addressed in a separate PR

Reply:
I've been looking into overriding the `tp_traverse` and `tp_clear` so we can have cycles with `__del__` and still deal with it. Not sure if that is something we can actually do, but...

Reply:
I think we could leave the `ContainerProxy` as a sort of `ContainerPointer` that only deals with the de-alloc, which would make the code a LOT simpler... if we removed `Stream.seek`.

(Of the top of my head at least).